### PR TITLE
Location model

### PR DESCRIPTION
**Pull request by Parmeshk7**

### CODE REVIEW

File: model/Location.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from flask_sqlalchemy import SQLAlchemy
+from User import Users
+db = SQLAlchemy()
+
+class Location(db.Model):
```
Comment (BharatShukla30) [inline model/Location.py:5]: Change class name to User_location, change file name too

File: model/User.py [hunk 2]
```diff
@@ -8,7 +8,7 @@ class Users(db.Model):
     email = db.Column(db.String(120), unique=True)
     password = db.Column(db.String(120))
     blood_group = db.Column(db.String(3))
-    location = db.Column(db.String(100))
+    # location = db.Column(db.String(100))
```
Comment (BharatShukla30) [inline model/User.py:11]: Remove this completely no need to comment this code

File: main.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 from werkzeug import security
 from constants import ARCGIS_URL, ARCGIS_API_KEY
 from model.User import db, Users
+from model.Location import db, Location
```
Comment (BharatShukla30) [inline main.py:8]: Change accordingly

File: main.py [hunk 4]
```diff
@@ -54,17 +55,28 @@ def register():
         user_type = request.json.get('user_type')
         location = request.json.get('location')
         password = request.json.get('password')
+
+        #####   Added the City and State Attributes also
```
Comment (BharatShukla30) [inline main.py:59]: no need of comment here the variables name are sufficient

File: main.py [hunk 5]
```diff
@@ -54,17 +55,28 @@ def register():
...
 
         if user is not None:
             response = jsonify({'message': 'Registration Failed, user already exists'})
             response.status_code = 401
         else:
             user = Users(name=name, email=email, password=hash_password, blood_group=blood_group, user_type=user_type,
                          location=location)
+            user_location = Location(city = city, state = state, location = location)
+
+            #Adding the Entries into the db
```
Comment (BharatShukla30) [inline main.py:77]: remove comment, not needed code self explainatory.

File: main.py [hunk 6]
```diff
@@ -97,6 +110,11 @@ def get_nearest_users(user_id, d):
         if distance <= d:
             nearest_users.append({'id': u.id, 'distance': distance, 'latitude': nearest_user_latitude,
                                   'longitude': nearest_user_longitude})
+            
+        latitude, longitude = user.location.split(', ')
+        nearest_user_latitude = u.location.split(', ')[0]
```
Comment (BharatShukla30) [inline main.py:115]: what is the change here?

**Review by BharatShukla30 — CHANGES_REQUESTED** [review]

File: model/Location.py (hunk 1 above)
Comment (Parmeshk7) [reply model/Location.py:5]: Done

File: model/User.py (hunk 2 above)
Comment (Parmeshk7) [reply model/User.py:11]: Done

File: main.py (hunk 3 above)
Comment (Parmeshk7) [reply main.py:8]: Done

File: main.py (hunk 4 above)
Comment (Parmeshk7) [reply main.py:59]: Done

File: main.py (hunk 5 above)
Comment (Parmeshk7) [reply main.py:77]: Done

File: main.py (hunk 6 above)
Comment (Parmeshk7) [reply main.py:115]: Actually I am not sure about the value that is stored in location variable, like how it actually representing the coordinates.

**Comment by BharatShukla30** [comment]
Not needed anymore

**Pull request closed without merging**